plugin-health-scoring/fetch-report.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+#
+# MIT License
+#
+# Copyright (c) 2024 Jenkins Infra
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+## Check for requirements
+for cli in curl jq mkdir cp
+do
+    command -v "${cli}" || { echo "ERROR: command $cli is missing. Exiting."; exit 1; }
+done
+
+: "${REPORTS_FOLDER?Environment variable 'REPORTS_FOLDER' unset}"
+: "${ETAGS_FILE?Environment variable 'REPORTS_FOLDER' unset}"
+: "${REPORT_FILE?Environment variable 'REPORTS_FOLDER' unset}"
+
+# Pre-check for etags
+curl --location --silent --show-error --remote-name "https://reports.jenkins.io/${REPORTS_FOLDER}/${ETAGS_FILE}" || echo "No previous etags file."
```

Comment:
Does this curl call fails only when etags file is missing?

Reply:
Good catch. It can also fail on HTTP errors.

I'm not sure what is "etag" and what is the purpose of this command.

Ping @alecharp could you help us? What was the goal of this?

Reply:
Sorry I'm late on this. The etag serves to not download the json file if the etag is the same. The etag I put on the header is the date time (seconds since epoch) of the latest score computed.